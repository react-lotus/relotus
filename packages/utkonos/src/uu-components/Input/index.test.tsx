/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from '.';

const onChange = jest.fn();

describe('UIKIT | Input NEW', () => {
  describe('Snapshots', () => {
    it('default', () => {
      const tree = renderer.create(<Input name="testName" onChange={onChange} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('labeled row', () => {
      const tree = renderer
        .create(<Input name="testName" label="labeled" onChange={onChange} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('labeled column', () => {
      const tree = renderer
        .create(<Input name="testName" label="labeled" direction="column" onChange={onChange} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('with error', () => {
      const tree = renderer
        .create(<Input name="testName" error="Test error" onChange={onChange} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('with suffix', () => {
      const tree = renderer
        .create(<Input name="testName" suffix="Руб" onChange={onChange} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled', () => {
      const tree = renderer.create(<Input name="testName" disabled onChange={onChange} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled with suffix', () => {
      const tree = renderer
        .create(<Input name="testName" suffix="Руб" disabled onChange={onChange} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled with suffix and Error', () => {
      const tree = renderer
        .create(
          <Input name="testName" suffix="Руб" disabled error="Test error" onChange={onChange} />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
