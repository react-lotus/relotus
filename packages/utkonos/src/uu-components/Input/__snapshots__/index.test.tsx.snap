// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UIKIT | Input NEW Snapshots default 1`] = `
<div
  className="uu-inputBox"
>
  <input
    className="uu-input"
    id="nanoid"
    name="testName"
    onChange={[MockFunction]}
  />
</div>
`;

exports[`UIKIT | Input NEW Snapshots disabled 1`] = `
<div
  className="uu-inputBox"
>
  <input
    className="uu-input"
    disabled={true}
    id="nanoid"
    name="testName"
    onChange={[MockFunction]}
  />
  <div
    className="uu-fieldAddon uu-inputAddon"
  >
    <svg
      className="uu-fieldAddonDisabled"
    >
      lock.svg
    </svg>
  </div>
</div>
`;

exports[`UIKIT | Input NEW Snapshots disabled with suffix 1`] = `
<div
  className="uu-inputBox"
>
  <input
    className="uu-input"
    disabled={true}
    id="nanoid"
    name="testName"
    onChange={[MockFunction]}
  />
  <div
    className="uu-fieldAddon uu-inputAddon"
  >
    <span
      className="uu-fieldAddonSuffix"
    >
      Руб
    </span>
    <svg
      className="uu-fieldAddonDisabled"
    >
      lock.svg
    </svg>
  </div>
</div>
`;

exports[`UIKIT | Input NEW Snapshots disabled with suffix and Error 1`] = `
<div
  className="uu-inputBox"
>
  <input
    className="uu-input uu-input_hasError"
    disabled={true}
    id="nanoid"
    name="testName"
    onChange={[MockFunction]}
  />
  <div
    className="uu-fieldAddon uu-inputAddon"
  >
    <span
      className="uu-fieldAddonSuffix"
    >
      Руб
    </span>
    <svg
      className="uu-fieldAddonDisabled"
    >
      lock.svg
    </svg>
  </div>
  <div
    className="uu-inputError"
  >
    <svg
      className="uu-inputErrorIcon uu-tooltipRoot uu-popoverRoot"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      error.svg
    </svg>
  </div>
</div>
`;

exports[`UIKIT | Input NEW Snapshots labeled column 1`] = `
<div
  className="uu-labeled_direction_column"
>
  <label
    aria-label="labeled"
    className="uu-labeledLabel_direction_column uu-label uu-label_text"
    htmlFor="nanoid"
  >
    labeled
  </label>
  <div
    className="uu-inputBox"
  >
    <input
      className="uu-input"
      id="nanoid"
      name="testName"
      onChange={[MockFunction]}
    />
  </div>
</div>
`;

exports[`UIKIT | Input NEW Snapshots labeled row 1`] = `
<div
  className="uu-labeled_direction_column"
>
  <label
    aria-label="labeled"
    className="uu-labeledLabel_direction_column uu-label uu-label_text"
    htmlFor="nanoid"
  >
    labeled
  </label>
  <div
    className="uu-inputBox"
  >
    <input
      className="uu-input"
      id="nanoid"
      name="testName"
      onChange={[MockFunction]}
    />
  </div>
</div>
`;

exports[`UIKIT | Input NEW Snapshots with error 1`] = `
<div
  className="uu-inputBox"
>
  <input
    className="uu-input uu-input_hasError"
    id="nanoid"
    name="testName"
    onChange={[MockFunction]}
  />
  <div
    className="uu-inputError"
  >
    <svg
      className="uu-inputErrorIcon uu-tooltipRoot uu-popoverRoot"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      error.svg
    </svg>
  </div>
</div>
`;

exports[`UIKIT | Input NEW Snapshots with suffix 1`] = `
<div
  className="uu-inputBox"
>
  <input
    className="uu-input"
    id="nanoid"
    name="testName"
    onChange={[MockFunction]}
  />
  <div
    className="uu-fieldAddon uu-inputAddon"
  >
    <span
      className="uu-fieldAddonSuffix"
    >
      Руб
    </span>
  </div>
</div>
`;
